### PR TITLE
Fix return statements #156

### LINKED ISSUE #156: QuantStamp Audit: QSP-7 - Missing return values
[QuantStamp security assessment](https://github.com/lidofinance/lido-dao/files/5561204/QuantStampReport.pdf) of [v0.1.0-rc.1](https://github.com/lidofinance/lido-dao/releases/tag/v0.1.0-rc.1) raised the following security issue:

# Issue: QSP-7

| Short Descr. | Missing return values |
|-------------|--------|
| Severity | Informational |
| Files affected | `DePool.sol`, `StakingProvidersRegistry.sol` |

## Description

The following functions are declared to return values but have no explicit return statement:

1. StakingProvidersRegistry.addStakingProvider does not return uint256 id .

2. DePool.submit is declared to return uint256 StETH but has no return statement.

## Recommendations

Add return statements to these functions.

### PR DESCRIPTION
**Pull request by lxzrv**
Closes #156
- Lido.submit() now returns shares amount generated instead of token amount because this value is fixed and carries more information than dynamically changing balance. if we return anything at all, that would be better.

### CODE REVIEW

File: contracts/0.4.24/Lido.sol [hunk 1]
```diff
@@ -123,9 +123,9 @@ contract Lido is ILido, IsContract, Pausable, AragonApp {
     /**
     * @notice Send funds to the pool with optional _referral parameter
     * @dev This function is alternative way to submit funds. Supports optional referral address.
-    * @return StETH Amount of StETH tokens generated
+    * @return Amount of StETH shares generated
```
Comment (ongrid) [inline contracts/0.4.24/Lido.sol:126]: Explain why shares instead of tokens. We discussed this verbally, but this should be stated explicitly in commit message.

File: contracts/0.4.24/Lido.sol [hunk 2]
```diff
@@ -123,9 +123,9 @@ contract Lido is ILido, IsContract, Pausable, AragonApp {
...
     */
-    function submit(address _referral) external payable returns (uint256 StETH) {
+    function submit(address _referral) external payable returns (uint256) {
```
Comment (ongrid) [inline contracts/0.4.24/Lido.sol:128]: this behavior should be covered by test and returned amount should be asserted.

File: contracts/0.4.24/nos/NodeOperatorsRegistry.sol [hunk 3]
```diff
@@ -111,6 +111,8 @@ contract NodeOperatorsRegistry is INodeOperatorsRegistry, IsContract, AragonApp
         operator.stakingLimit = _stakingLimit;
 
         emit NodeOperatorAdded(id, _name, _rewardAddress, _stakingLimit);
+
+        return id;
```
Comment (ongrid) [inline contracts/0.4.24/nos/NodeOperatorsRegistry.sol:115]: This is out of scope. Should be done in a separate issue and in separate commit.
I'd propose to drop this change.

**Review by ongrid — CHANGES_REQUESTED** [review]
* The new behavior must be tested, numbers must be asserted.
* Changing returnable value should be justified in commit message.

File: contracts/0.4.24/Lido.sol (hunk 1 above)
Comment (lxzrv) [reply contracts/0.4.24/Lido.sol:126]: Done

File: contracts/0.4.24/Lido.sol (hunk 2 above)
Comment (lxzrv) [reply contracts/0.4.24/Lido.sol:128]: Discussed verbally, tests will be later.

Comment (ongrid) [reply contracts/0.4.24/Lido.sol:128]: Reason: return value of transact is not visible in the returned object of promise. To test this we need to wrap into `.call()`able mock but we are under time pressure. We'll put this in backlog.

**Review by ongrid — APPROVED** [review]

**Review by krogla — APPROVED** [review]
looks good
